pkg/auth/oidc.go

```diff
@@ -0,0 +1,109 @@
+package auth
+
+import (
+	"context"
+	"strings"
+
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/logrus/ctxlogrus"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/idtoken"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+var InvalidAuthError = status.Errorf(codes.Unauthenticated, "invalid authentication credentials")
+
+type Validator interface {
+	Validate(ctx context.Context, token string, audience string) (*idtoken.Payload, error)
+}
+
+type OidcValidator struct {
```

Comment:
Calling this as `OIDCValidator` for acronym would be more suitable according to [this](https://github.com/golang/go/wiki/CodeReviewComments#initialisms)

Reply:
that's right, i've modified it, thanks!